# Analyzing   and determining whether there is any difference in average TAT among the different laboratories at 5% significance level.
 


In [93]:
import pandas as pd
import numpy as np
import matplotlib.pylab as plt

In [94]:
data=pd.read_csv("labTAT.CSV")

In [95]:
data.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [96]:
data.shape

(120, 4)

In [97]:
data.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [98]:
data.columns

Index(['Laboratory 1', 'Laboratory 2', 'Laboratory 3', 'Laboratory 4'], dtype='object')

# ANOVA TEST 

In [99]:
### Statistical Normality Tests
# Shapiro-Wilk Test
from numpy.random import seed
from numpy.random import randn
from scipy.stats import shapiro


In [100]:
# normality test
stat, p = shapiro(data)

In [101]:
print('Statistics=%.3f, p=%.3f' % (stat, p))

Statistics=0.995, p=0.118


#  H0= Lab reports is in Normal,Ha= Lab Reports is not in Normal
# p value is 0.118,p>0.05, P is high null fly, Accpet null hypothesis, What null hypothesis says Laboratory data is normal

In [102]:
from scipy import stats

In [103]:
stats.f_oneway(data["Laboratory 1"],data["Laboratory 2"],data["Laboratory 3"],data["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

# P value=2.1156708949992414e-57,p<0.05,P low , fail to accept null hypothesis , what alternate hypothesis says "Average turn around time of all 4 laboratories are not same"
# CONCLUSION: Hence statistically Proved that Average turn around time of all 4 laboratories are not same.




# CULTLET Data SetProblem Statement  ####################

# Determining whether any difference of cutlet diameters of two units checking by hypothesis

In [104]:
cut=pd.read_csv("Cutlets.csv")

In [105]:
cut.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [106]:
cut.isnull().sum()

Unit A    0
Unit B    0
dtype: int64

In [107]:
cut.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [108]:
cut.columns

Index(['Unit A', 'Unit B'], dtype='object')

# first check the data is normally distributed or not by using normality test 

In [109]:
# normality test
stat, p = shapiro(cut)

In [110]:
print('Statistics=%.3f, p=%.3f' % (stat, p))

Statistics=0.976, p=0.204


# P= 0.204,P>0.05, P high Null fly, Accept Null Hypothesis , what Null hypothesis says "cutlet data is normal ", hence Statistically Proved


# variance test 
# H0=(Variance)unitA=(Variance)unitB
# (Variance)unita=!(variance)unitB

In [111]:
c=pd.DataFrame(cut)

In [112]:

import statistics
from statistics import variance 

In [113]:
c

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [114]:
c=c.rename(columns={"Unit A":"sample1","Unit B":"sample2"})

In [133]:
c.describe()

sample1    sample2
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [115]:
c

sample1  sample2
0    6.8090   6.7703
1    6.4376   7.5093
2    6.9157   6.7300
3    7.3012   6.7878
4    7.4488   7.1522
5    7.3871   6.8110
6    6.8755   7.2212
7    7.0621   6.6606
8    6.6840   7.2402
9    6.8236   7.0503
10   7.3930   6.8810
11   7.5169   7.4059
12   6.9246   6.7652
13   6.9256   6.0380
14   6.5797   7.1581
15   6.8394   7.0240
16   6.5970   6.6672
17   7.2705   7.4314
18   7.2828   7.3070
19   7.3495   6.7478
20   6.9438   6.8889
21   7.1560   7.4220
22   6.5341   6.5217
23   7.2854   7.1688
24   6.9952   6.7594
25   6.8568   6.9399
26   7.2163   7.0133
27   6.6801   6.9182
28   6.9431   6.3346
29   7.0852   7.5459
30   6.7794   7.0992
31   7.2783   7.1180
32   7.1561   6.6965
33   7.3943   6.5780
34   6.9405   7.3875

In [118]:
print("Variance of sample set is % s" 
      %(statistics.variance(c.sample1)))

Variance of sample set is 0.08317945374789917


In [119]:
print("Variance of sample set is % s" 
      %(statistics.variance(c.sample2)))

Variance of sample set is 0.11792400440336137


# P Value for my data set greater than 0.05,P high Null fly , Accept Null Hypothesis, what Null Hypothesus says is Variance of sample1 = variance Of sample2


# 2 Smple T Test 

In [122]:
stats.ttest_ind(c.sample2,c.sample1)

Ttest_indResult(statistic=-0.7228688704678061, pvalue=0.47223947245995)

# P=0.4722,P>0.05,P High Null Fly, Accept Null hypothesis , what Null hypothesis says "(Mean)sample1 = (Mean)sample2
# CONCLUSION: Hence Statistically Proved That  Cutlet diameters Of Both Unit A & unit B are Equal There Is No Difference like this we will say to the Managers or clients 


# Fantaloons Sales managers commented that % of males versus females walking in to the store differ based on day of the week

In [125]:
f=pd.read_csv("Faltoons.csv")

In [126]:
f.head()

Weekdays Weekend
0     Male  Female
1   Female    Male
2   Female    Male
3     Male  Female
4   Female  Female

In [127]:
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
count = np.array([66,47])
nobs = np.array([167,120])
stat, pval = proportions_ztest(count, nobs)
print('{0:0.3f}'.format(pval))

0.952


# p = 0.952,p>0.05, P high Null fly, Acccept Null hypothesis, what null Hypothesis says is "equal Proportions"
# CONCLUSION : sales manager commit to start fantaloons sales at weekends and weekdays both , why Because Both The days sales with respect to male Vs Female is equal Proportions, statistically proved


# Buyers ratio Data set

In [161]:
# H0= All the Averages are Equal
# Ha= Atleast One Is different From Other

In [162]:
br=pd.read_csv("BuyerRatio (1).csv")

In [163]:
br.head()

Observed Values  East  West  North  South
0                1    50   142    131     70
1                0   435  1523   1356    750

In [170]:
br.describe()

Observed Values        East         West        North       South
count         2.000000    2.000000     2.000000     2.000000    2.000000
mean          0.500000  242.500000   832.500000   743.500000  410.000000
std           0.707107  272.236111   976.514465   866.205807  480.832611
min           0.000000   50.000000   142.000000   131.000000   70.000000
25%           0.250000  146.250000   487.250000   437.250000  240.000000
50%           0.500000  242.500000   832.500000   743.500000  410.000000
75%           0.750000  338.750000  1177.750000  1049.750000  580.000000
max           1.000000  435.000000  1523.000000  1356.000000  750.000000

In [164]:
from scipy.stats import chisquare

In [165]:
chisquare(br.East)

Power_divergenceResult(statistic=305.6185567010309, pvalue=1.9665918567457293e-68)

In [166]:
chisquare(br.West)

Power_divergenceResult(statistic=1145.442042042042, pvalue=4.390318990857786e-251)

In [167]:
chisquare(br.North)

Power_divergenceResult(statistic=1009.1627437794217, pvalue=1.830731624970013e-221)

In [168]:
chisquare(br.South)

Power_divergenceResult(statistic=563.9024390243902, pvalue=1.1904536774684822e-124)

# Here P Value is <0.05  so P High null go, fail to accept Null hypothesis what alternate  hypothesis says is
# Proportion Of buyersratio Across all the Regions Should NOT Be Same Statistically Proved #####
